src/ucp/rndv/rndv_ppln.c

```diff
@@ -247,8 +274,16 @@ static ucs_status_t ucp_proto_rndv_ppln_progress(uct_pending_req_t *uct_req)
             return UCS_OK;
         }
 
+        /* Avoid sending message less than `frag_proto_min_length` */
+        next_frag = rpriv->frag_size;
+        msg_left  = req->send.state.dt_iter.length - req->send.state.dt_iter.offset;
+        if ((msg_left > rpriv->frag_size) &&
+            (msg_left < rpriv->frag_size + rpriv->frag_proto_min_length)) {
+            next_frag = rpriv->frag_proto_min_length;
+        }
```

Comment:
is it an unrelated fix?

Reply:
No, it is related, let me explain the case. 

In the master we have `ucp_proto_select_elem_t` which contains protocols for every message size. Now we have only one proto which has it's own `min_length`.

Lets imagine that we are sending 300 bytes using RNDV PPLN proto and the frag size is 130 bytes. Then if we use master logic, we send two 130 bytes fragments using RNDV-get and one 40 bytes remainder (300 - 130*2) using RNDV-put since RNDV-get cannot send such small chunks (min_length of RNDV-get is 129).

In the new approach we don't have ability to choose different proto for small remainders. So I solved this problem by sending min_length first, so the next fragment would leave a remainder at all. 

It also can be done by increasing the current fragment, but we need to care about max supported message length for current proto in that case.